lib/auth/clt.go

```diff
@@ -149,58 +171,35 @@ func NewClient(cfg client.Config, params ...roundtrip.ClientParam) (*Client, err
 		},
 		params...,
 	)
-	roundtripClient, err := roundtrip.NewClient("https://"+teleport.APIDomain, CurrentVersion, clientParams...)
+	httpClient, err := roundtrip.NewClient("https://"+teleport.APIDomain, CurrentVersion, clientParams...)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	return &Client{
-		APIClient: *apiClient,
-		Client:    *roundtripClient,
+	return &HTTPClient{
+		Client:    *httpClient,
 		transport: transport,
-		TLS:       tlsConfig,
+		tls:       tls,
 	}, nil
 }
 
-// EncodeClusterName encodes cluster name in the SNI hostname
-func EncodeClusterName(clusterName string) string {
-	// hex is used to hide "." that will prevent wildcard *. entry to match
-	return fmt.Sprintf("%v.%v", hex.EncodeToString([]byte(clusterName)), teleport.APIDomain)
-}
-
-// DecodeClusterName decodes cluster name, returns NotFound
-// if no cluster name is encoded (empty subdomain),
-// so servers can detect cases when no server name passed
-// returns BadParameter if encoding does not match
-func DecodeClusterName(serverName string) (string, error) {
-	if serverName == teleport.APIDomain {
-		return "", trace.NotFound("no cluster name is encoded")
-	}
-	const suffix = "." + teleport.APIDomain
-	if !strings.HasSuffix(serverName, suffix) {
-		return "", trace.NotFound("no cluster name is encoded")
+// Close closes the HTTP client connection to the auth server.
+func (c *HTTPClient) Close() {
+	c.mux.Lock()
+	defer c.mux.Unlock()
+	if c.transport != nil {
+		c.transport.CloseIdleConnections()
 	}
-	clusterName := strings.TrimSuffix(serverName, suffix)
+}
 
-	decoded, err := hex.DecodeString(clusterName)
-	if err != nil {
-		return "", trace.BadParameter("failed to decode cluster name: %v", err)
-	}
-	return string(decoded), nil
+// TLSConfig returns the HTTP client's TLS config.
+func (c *HTTPClient) TLSConfig() *tls.Config {
+	return c.tls
 }
 
-// ClientTimeout sets idle and dial timeouts of the HTTP transport
-// used by the client.
-func ClientTimeout(timeout time.Duration) roundtrip.ClientParam {
-	return func(c *roundtrip.Client) error {
-		transport, ok := (c.HTTPClient().Transport).(*http.Transport)
-		if !ok {
-			return nil
-		}
-		transport.IdleConnTimeout = timeout
-		transport.ResponseHeaderTimeout = timeout
-		return nil
-	}
+// GetTransport returns the HTTP client's transport.
+func (c *HTTPClient) GetTransport() *http.Transport {
+	return c.transport
```

Comment:
Not needed?

Reply:
I'm not sure, since it's exported it may already be used outside of Teleport. I embedded `HTTPClient` into `Client` and preserved this method as is. The whole HTTPClient should be deprecated by 8.0, so I wouldn't worry about it.